### ISSUE TITLE
Project level 7

### ISSUE THREAD

**Issue by E0857389** [nus-cs2030/2122-s2#325]
hello i dont really understand how the probability thing works for the rest time. does that mean that whenever the supplier is called a new rest time would be given?

**Comment by alantay11** [comment]
yes when its called the rng will give a new rest time which can be 0 or some random number depending on the probability so you don't need to take care of it. the seed for the rng is always the same so the 'random' numbers won't be random in the output

**Comment by E0857389** [comment]
> 

hello! do u have any hints on level7? i tried calling the supplier get only in the done event and updated the servers avail time by adding the rest time at the done event. is there a flaw with my logic?

**Comment by alantay11** [comment]
i think if you do everything in the done event you won't be able to make the server free?

**Comment by yizziepeasy** [comment]
> i think if you do everything in the done event you won't be able to make the server free?

Did you have a boolean to tag each server as Resting?

**Comment by alantay11** [comment]
nope but i have a boolean to check if they're idle

**Comment by Yang990121** [comment]
its the same as level 5, dont need care about the probability, the main class will settle for you.
Just treat it to be the same as level 5
